src/flp/types.rs

```diff
@@ -154,8 +176,18 @@ impl<F: FftFriendlyFieldElement> Type for Sum<F> {
     type Field = F;
 
     fn encode_measurement(&self, summand: &F::Integer) -> Result<Vec<F>, FlpError> {
-        let v = F::encode_as_bitvector(*summand, self.bits)?.collect();
-        Ok(v)
+        if summand > &self.max_measurement {
+            return Err(FlpError::Encode(format!(
+                "unexpected measurement: got {:?}; want ≤{:?}",
+                summand, self.max_measurement
+            )));
+        }
+
+        let enc_summand: Vec<F> = F::encode_as_bitvector(*summand, self.bits)?.collect();
+        let enc_summand_plus_offset =
+            F::encode_as_bitvector(self.offset + *summand, self.bits)?.collect();
+
+        Ok([enc_summand, enc_summand_plus_offset].concat())
```

Comment:
This will save two allocations and two memcpys:
```suggestion
        let mut encoded = Vec::with_capacity(self.bits * 2);
        let enc_summand = F::encode_as_bitvector(*summand, self.bits)?;
        encoded.extend(enc_summand);
        let enc_summand_plus_offset = F::encode_as_bitvector(self.offset + *summand, self.bits)?;
        encoded.extend(enc_summand_plus_offset);

        Ok(encoded)
```